ggml/src/ggml-rpc.cpp

```diff
@@ -279,6 +308,101 @@ static bool send_rpc_cmd(const std::shared_ptr<socket_t> & sock, enum rpc_cmd cm
     return true;
 }
 
+static bool send_request_alloc_buffer(const std::shared_ptr<socket_t> & sock, const request_alloc_buffer & request, response_alloc_buffer & response) {
+    // serialization format: | size (8 bytes) |
+    int input_size = sizeof(uint64_t);
+    std::vector<uint8_t> input(input_size, 0);
+    memcpy(input.data(), &request.size, sizeof(request.size));
+    std::vector<uint8_t> output;
+    bool status = send_rpc_cmd(sock, RPC_CMD_ALLOC_BUFFER, input, output);
+    if (!status) {
+        return false;
+    }
+    GGML_ASSERT(output.size() == 2*sizeof(uint64_t));
+    // output serialization format: | ptr (8 bytes) | size (8 bytes) |
+    memcpy(&response.ptr, output.data(), sizeof(response.ptr));
+    memcpy(&response.size, output.data() + sizeof(response.ptr), sizeof(response.size));
```

Comment:
I don't think this is much better than the previous version. The idea would be to read directly into the struct from the network.

Reply:
Thanks for looking into this, I get what you mean. Could you take another look and let me know if you agree?

I also noticed that all commands have a known output size, maybe we can skip sending output size from the server?

Reply:
Yep, that's exactly what I meant. 

> I also noticed that all commands have a known output size, maybe we can skip sending output size from the server?

I don't have a strong opinion about this either way. Some commands may have variable output size in the future, for example `get_description` in the device interface, and having the output size may help simplify the code a bit in that case, but it is probably not very important either way.